sstables/sstable_set.hh

```diff
@@ -121,6 +122,7 @@ public:
         });
     }
     void insert(shared_sstable sst);
+    void insert(const std::vector<shared_sstable>& ssts);
```

Comment:
std::span is nicer for such interfaces.

Reply:
makes sense.

Reply:
@avikivity Has our ban on c++20 ranges been lifted yet?

Reply:
std::span is not part of std::ranges.

The ban on std::ranges will be lifted when we move to clang 16, which fixed a clang bug that caused many uses of std::ranges to fail.